tfsdk/config.go

```diff
@@ -21,10 +21,43 @@ func (c Config) Get(ctx context.Context, target interface{}) diag.Diagnostics {
 	return reflect.Into(ctx, c.Schema.AttributeType(), c.Raw, target, reflect.Options{})
 }
 
-// GetAttribute retrieves the attribute found at `path` and returns it as an
+// GetAttribute retrieves the attribute found at `path` and populates the
+// `target` with the value.
+func (c Config) GetAttribute(ctx context.Context, path *tftypes.AttributePath, target interface{}) diag.Diagnostics {
+	attrValue, diags := c.getAttributeValue(ctx, path)
+
+	if attrValue == nil {
+		return diags
+	}
+
+	valueAsDiags := ValueAs(ctx, attrValue, target)
+
+	// ValueAs does not have path information for its Diagnostics.
+	for idx, valueAsDiag := range valueAsDiags {
+		if valueAsDiag.Severity() == diag.SeverityError {
+			valueAsDiags[idx] = diag.NewAttributeErrorDiagnostic(
+				path,
+				valueAsDiag.Summary(),
+				valueAsDiag.Detail(),
+			)
+		} else if valueAsDiag.Severity() == diag.SeverityWarning {
+			valueAsDiags[idx] = diag.NewAttributeWarningDiagnostic(
+				path,
+				valueAsDiag.Summary(),
+				valueAsDiag.Detail(),
+			)
+		}
```

Comment:
Eek. Can we add an issue at least to have `diag.Diagnostic` have a `SetPath` method to avoid this kind of stuff?

Reply:
I created https://github.com/hashicorp/terraform-plugin-framework/issues/169, but it might require some refactoring to the `diag` package.